vowpalwabbit/OjaNewton.cc

```diff
@@ -52,51 +52,53 @@ struct OjaNewton {
     bool normalize;
     bool random_init;
 
-    void initialize_Z()
-    {
-        size_t stride_shift = all->reg.stride_shift;
-        weight* weights = all->reg.weight_vector;
-
-        uint32_t length = 1 << all->num_bits;
-        if(normalize) { // initialize normalization part
-	    for (uint32_t i = 0; i < length; i++)
-	        weights[(i << stride_shift) + NORM2] = 0.1;
-        }
-
+	void initialize_Z()
+	{  weight_vector& weights = all->wv;
+       uint32_t length = 1 << all->num_bits;
+	   if (normalize) { // initialize normalization part
+		 for (weight_vector::iterator i = weights.begin(NORM2); i != weights.end(NORM2); ++i)
```

Comment:
Here's an example of the interface suggestion I had for using a range.  

    for (auto & ww: weights.mutate(NORM2))
    {
      ww = 0.1;
    }

Seems a bit nicer, no?

In any case, your should avoid writing loops like these.  The above is really just 

`std::fill(weights.begin(NORM2), weights.end(NORM2), 0.1)`.

Or, even better, if you decide to use boost.range library, along with the range-based interface I suggested:

`boost::fill(weights.mutate(NORM2), 0.1);